src/cls/rgw/cls_rgw.cc

```diff
@@ -707,10 +707,24 @@ int rgw_bucket_prepare_op(cls_method_context_t hctx, bufferlist *in, bufferlist
   return write_bucket_header(hctx, &header);
 }
 
+static bool obj_name_filter(string& name)
+{
+  int ret = true;
+  if (name.substr(0,10) == MP_MULTI_SUFFIX) {
```

Comment:
Like below "10" is magic, ugh.  also substr makes a temp copy - better to use compare.  Or better yet use "boost::algorithm::starts_with".